agent/src/agentclient.rs

```diff
@@ -398,41 +434,81 @@ impl<T: APIServerClient> BrupopAgent<T> {
                 }
             };
 
+            let bottlerocket_shadow_status = bottlerocket_shadow
+                .status
+                .as_ref()
+                .context(agentclient_error::MissingBottlerocketShadowStatus)?;
+
             match self.handle_state_transition(&bottlerocket_shadow).await {
-                Ok(()) => {}
+                Ok(()) => {
+                    let crash_count;
+                    let state_transition_failure_timestamp;
+                    match bottlerocket_shadow_status.current_state {
+                        // Reset crash_count and state_transition_failure_timestamp for a successful update loop
+                        BottlerocketShadowState::MonitoringUpdate => {
+                            crash_count = 0;
+                            state_transition_failure_timestamp = None;
```

Comment:
I think we'll want to wait to apply these until the time that we are transitioning from `MonitoringForUpdate` into `Idle`.

Reason being is that we want to be able to add support for customers to add custom checks that the node is healthy (#12, which references [this similar features in CLUO](https://github.com/coreos/container-linux-update-operator/blob/master/doc/before-after-reboot-checks.md)). Basically, we want any check that fails here to result the node being reverted to the previous version, and maintaining its crash loop state.

We can implement that functionality in a future release, but we should still make sure to only assign the fresh state after we've reached `Idle`.